python/plugins/processing/core/parameters.py

```diff
@@ -770,3 +770,41 @@ def dataType(self):
                 types += 'any, '
 
         return types[:-2]
+
+
+class ParameterPredicate(Parameter):
```

Comment:
IMO it will be better to use ParameterGeometryPredicate as we also might want to add other predicates in future, e.g. boolean or SQL